### DISCUSSION TITLE
Issue with some molecules

### DISCUSSION THREAD

**Discussion by mearcstapa-gqz** [snap-stanford/ogb#199] [PCQM4M-LSC]
Hi, Is it intended that
1) There are metals (with few occurrences) in this dataset.
2) There are unseen elements in test set.
3) There are few thousands molecules with broken side chain(s) in this dataset, such as [CH]NC(=O)C=C(C)C, [NH]CCC[NH], [CH]NCCc1cccnc1, etc.

**Comment by weihua916** [comment]
Hi! Great questions!

The dataset is what it is, so 1 and 3 could happen. 2 is probably a by-product of our data split. Given that unseen atoms are extremely rare in the test set (0.0008% of test atoms are unseen during training, 0.01% of test molecules contain unseen atoms), it should not affect the result much (e.g., even if you make a big 10 eV error on those molecules, it will only affect the entire test MAE by 0.001). Properly handling unseen atoms (e.g., make sure the prediction is not extreme, using periodic table information rather than atom ID as input features) might help very slightly.

Hope this helps!

**Reply by mearcstapa-gqz** [reply]
Hi! Thanks! Another question though, I'm curious about the SMILES generation process. (Because I can't reverse-engineer the CID of some SMILES, lol.) In the homepage of PubChemQC project, http://pubchemqc.riken.jp/, there says "Only for the singlet and neutral molecules". 
However, there're charged molecules in PCQM4M-LSC dataset.
And also, for the carbenes (nitrenes) mentioned in 3, except for those carbenes with an atom with lone pair connected to it, the ground state for those carbenes are most likely to be triplet.

**Reply by weihua916** [reply]
Hi! It's rather complicated, so I suggest you not to reverse-engineer :)
The dataset and the task are both correct and chemically meaningful, as confirmed by our team member Maho Nakata (lead author of PubChemQC project). If you have further chemistry questions, please reach out to him.

**Reply by mearcstapa-gqz** [reply]
Cool, I would correspond with him hereafter on chemistry questions. Thanks! Just wanna make sure, since there are around 300 charged molecules in PCQM4M-LSC, but both the PubChemQC paper and project homepage claim that the all molecules are neutral.

**Reply by weihua916** [reply]
Great point. I remember there was some discussion about the charges when we created the dataset. It'd be helpful for you to summarize your correspondence in this thread so that the community can also understand the design choice.

**Reply by mearcstapa-gqz** [reply]
Hi, sorry for the late reply. Not sure about what you mean by "It'd be helpful for you to summarize your correspondence in this thread". Does it means that I summarize some characteristics of the dataset, namely there exists 

- metals (in the form of element, inorganic salt, organometallic compound) (with few occurrences, not exceeding fourth period) in this dataset
- unseen elements in test set
- few thousand singlet and triplet carbenes/ nitrenes in this dataset
- few hundred charged molecules in this dataset
- several thousand duplicate SMILES strings in this dataset

**Reply by weihua916** [reply]
Thanks. I meant summarizing your correspondence with Maho, which hopefully resolves your concerns.

**Reply by weihua916** [reply]
Hi! Great point. We are aware that the same 2D molecular graph could give different HOMO-LUMO gaps, and that's because of multiple local minima in the molecule geometry optimization. In other words, there could be multiple locally-optimum 3D geometries for the same 2D SMILES graph, which results in the different HOMO-LUMO gaps. 

Also, there are some subtleties in our SMILES calculation strategy and how we handle formal charges etc. Please consult our member, Maho Nakata (http://nakatamaho.riken.jp/, maho.nakata@gmail.com) for more expert answers.
